### DISCUSSION TITLE
Request for an Official Redux Integration Example

### DISCUSSION THREAD

**Discussion by shxntanu** [xyflow/xyflow#4289] [New Features]
Hello ReactFlow Team,

I have been working on integrating ReactFlow into a project that uses Redux for state management. While I found the advanced section in the documentation, specifically the example with Zustand, very useful, I noticed that there isn't an official example demonstrating integration with Redux. Given Redux's popularity and widespread use in the React ecosystem, having an official example would be extremely beneficial for many developers, including myself.

During my integration process, I encountered several challenges due to the lack of resources and examples on how to effectively combine ReactFlow with Redux. After spending considerable time looking for solutions and piecing together information from various sources, I managed to get it working. However, I believe that having an official example would save others from facing similar hurdles and make ReactFlow even more accessible.

Here are some specific points where an official Redux example could provide clarity:

- Examples of handling ReactFlow actions within Redux action creators and reducers.
- Example on creating a slice within Redux and then including the correct methods from the library to update the state in the store
- Handling dragging of nodes on the canvas and registering their correct position using Redux
- A complete example with draggable nodes, drag and drop and editable nodes using Redux and its best practices.

By providing a detailed example, you would not only help those who are already familiar with Redux but also attract new users to ReactFlow by showcasing its versatility and ease of integration with different state management libraries.

Thank you for considering this request. I am happy to contribute to the discussion and provide any additional information or examples from my experience if needed.

Best regards,
Shantanu Wable

**Comment by bcakmakoglu** [comment]
There might not be an example for redux on the docs itself but there is one in the repo.

[Check this link.](https://github.com/xyflow/xyflow/tree/main/examples/react/src/examples/Redux)

**Reply by shxntanu** [reply]
Oh alright, Thank you!
